codec/src/main/java/io/netty/handler/codec/compression/ZstdEncoder.java

```diff
@@ -0,0 +1,194 @@
+     */
+    public ZstdEncoder() {
+        this(DEFAULT_COMPRESSION_LEVEL, DEFAULT_BLOCK_SIZE, MAX_BLOCK_SIZE);
+    }
+
+    /**
+     * Creates a new Zstd encoder.
+     *  @param  compressionLevel
+     *            specifies the level of the compression
+     */
+    public ZstdEncoder(int compressionLevel) {
+        this(compressionLevel, DEFAULT_BLOCK_SIZE, MAX_BLOCK_SIZE);
+    }
+
+    /**
+     * Creates a new Zstd encoder.
+     *  @param  blockSize
+     *            is used to calculate the compressionLevel
+     *  @param  maxEncodeSize
+     *            specifies the size of the largest compressed object
+     */
+    public ZstdEncoder(int blockSize, int maxEncodeSize) {
+        this(DEFAULT_COMPRESSION_LEVEL, blockSize, maxEncodeSize);
+    }
+
+    /**
+     * @param  blockSize
+     *           is used to calculate the compressionLevel
+     * @param  maxEncodeSize
+     *           specifies the size of the largest compressed object
+     * @param  compressionLevel
+     *           specifies the level of the compression
+     */
+    private ZstdEncoder(int compressionLevel, int blockSize, int maxEncodeSize) {
+        super(true);
+        this.compressionLevel = ObjectUtil.checkPositive(compressionLevel, "compressionLevel");
+        this.blockSize = blockSize;
+        this.maxEncodeSize = ObjectUtil.checkPositive(maxEncodeSize, "maxEncodeSize");
+    }
+
+    @Override
+    protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean preferDirect) {
+        return allocateByteBuffer(ctx, msg, preferDirect);
+    }
+
+    private ByteBuf allocateByteBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean preferDirect) {
+        if (!preferDirect) {
+            throw new EncoderException("dstBuff must be a direct buffer");
+        }
+
+        if (buffer == null) {
+            throw new IllegalStateException("not added to a pipeline," +
+                    "or has been removed,buffer is null");
+        }
+
+        long targetBufSize = 0;
+        int remaining = msg.readableBytes() + buffer.readableBytes();
+
+        // quick overflow check
+        if (remaining < 0) {
+            throw new EncoderException("too much data to allocate a buffer for compression");
+        }
+
+        while (remaining > 0) {
+            int curSize = Math.min(blockSize, remaining);
+            remaining -= curSize;
+            targetBufSize += Zstd.compressBound(curSize);
+        }
+
+        if (targetBufSize > maxEncodeSize || 0 > targetBufSize) {
+            throw new EncoderException("requested encode buffer size (" + targetBufSize + " bytes) exceeds " +
+                    "the maximum allowable size (" + maxEncodeSize + " bytes)");
+        }
+
+        int bufferSize = (int) targetBufSize;
+        return ctx.alloc().directBuffer(bufferSize, bufferSize);
+    }
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) {
+
```

Comment:
nit: remove

Reply:
I've done it

Reply:
I have removed the blank lines. Is there anything else I haven’t noticed here?